test/remote/gateways/remote_xpay_test.rb

```diff
@@ -47,4 +47,25 @@ def test_failed_purchase
   #   assert_success response
   #   assert_match 'PENDING', response.message
   # end
+
+  # def test_successful_refund ## test requires set up (purchase or auth through 3ds)
+  #   options = {
+  #     order: {
+  #       currency: 'EUR',
+  #       description: 'refund operation message'
+  #     },
+  #     operation_id: '168467730273233329'
+  #   }
+  #   response = @gateway.refund(@amount, options)
+  #   assert_success response
+  # end
+
+  # def test_successful_void ## test requires set up (purchase or auth through 3ds)
+  #   options = {
+  #     description: 'void operation message',
+  #     operation_id: '168467730273233329'
+  #   }
+  #   response = @gateway.void(@amount, options)
+  #   assert_success response
+  # end
```

Comment:
why is this commented on?

Reply:
in order to perform those kind of transactions a successful 3ds transaction should be executed and then void, refund or capture nevertheless this flow requires the previous setup done.